lib/Dialect/FIRRTL/Transforms/CheckCombCycles.cpp

```diff
@@ -579,6 +584,66 @@ void dumpSimpleCycle(SCCIterator &scc, FModuleOp module,
   }
 }
 
+// This function dumps a (shortest) path from `inputPort` to `outputPort` by
+// looking at a module referred by `instance`.
+void dumpPathBetweenModulePorts(SmallString<16> &instancePath,
+                                InstanceOp instance, unsigned inputPort,
+                                unsigned outputPort, NodeContext *context,
+                                mlir::InFlightDiagnostic &diag) {
+  auto module = dyn_cast<FModuleOp>(*instance.getReferencedModule());
+  if (!module)
+    return;
+  unsigned instancePathSize = instancePath.size();
+  instancePath += ".";
+  instancePath += instance.name();
+  auto start = Node(module.getArgument(inputPort), context);
+  auto end = Node(module.getArgument(outputPort), context);
+  llvm::DenseMap<Node, Node> previousNode;
+  // Find a shortest path from input port to output port by BFS.
+  std::queue<Node> que;
+  que.push(start);
+  while (!que.empty()) {
+    auto current = que.front();
```

Comment:
is current a value or a reference?  front returns a reference invalidated by pop.

Reply:
if you don't want to track the inference down, putting the pop at the end of the loop would certainly be safe.

Reply:
Thanks, addressed.